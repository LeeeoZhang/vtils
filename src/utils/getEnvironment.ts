import { inAndroid } from './inAndroid'
import { inBrowser } from './inBrowser'
import { inDeno } from './inDeno'
import { inIOS } from './inIOS'
import { inMiniProgram } from './inMiniProgram'
import { inNodeJS } from './inNodeJS'
import { inTaro } from './inTaro'
import { inWechatWebView } from './inWechatWebView'

export interface GetEnvironmentResult {
  /** 浏览器 */
  readonly browser: boolean
  /** 微信 WebView */
  readonly wechatWebView: boolean
  /** 小程序 */
  readonly miniProgram: boolean
  /** 微信小程序 */
  readonly wechatMiniProgram: boolean
  /** iOS */
  readonly ios: boolean
  /** 安卓 */
  readonly android: boolean
  /** Node.js */
  readonly nodejs: boolean
  /** Deno */
  readonly deno: boolean
  /** Taro 3 */
  readonly taro: boolean
}

let env: GetEnvironmentResult | undefined

/**
 * 获取运行环境信息。
 *
 * @returns 返回运行环境信息
 */
export function getEnvironment(): GetEnvironmentResult {
  if (env == null) {
    env = {
      browser: inBrowser(),
      wechatWebView: inWechatWebView(),
      miniProgram: !!inMiniProgram(),
      wechatMiniProgram: !!inMiniProgram('微信'),
      ios: inIOS(),
      android: inAndroid(),
      nodejs: inNodeJS(),
      deno: inDeno(),
      taro: inTaro(),
    }
  }
  return env
}
