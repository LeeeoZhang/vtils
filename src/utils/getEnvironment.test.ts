import { getEnvironment } from './getEnvironment'

describe('getEnvironment', () => {
  test('表现正常', () => {
    expect(getEnvironment()).toMatchSnapshot()
  })
})
