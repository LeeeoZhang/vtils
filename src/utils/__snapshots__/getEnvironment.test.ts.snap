// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEnvironment 表现正常 1`] = `
Object {
  "android": false,
  "browser": true,
  "deno": false,
  "ios": false,
  "miniProgram": false,
  "nodejs": true,
  "taro": false,
  "wechatMiniProgram": false,
  "wechatWebView": false,
}
`;
